cranelift/entity/src/lib.rs

```diff
@@ -160,13 +160,19 @@ macro_rules! entity_impl {
             }
 
             /// Return the raw bit encoding for this instance.
+            ///
+            /// Warning: entity types maintain a sentinel at `u32::MAX` and this
+            /// method may return it.
```

Comment:
Can we write something about this being an opaque encoding instead? (Currently it sounds like the underlying representation being mostly equal is a documented feature)

Something like: "The raw bit encoding is opaque and has no guaranteed correspondence to the entity's index. It encodes the entire state of this index value: either a valid index or an invalid-index sentinel. The value returned by this method should only be passed to `from_bits`."

and likewise below "should only be given bits from `as_bits`"